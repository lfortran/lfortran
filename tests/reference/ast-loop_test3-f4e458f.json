{
    "basename": "ast-loop_test3-f4e458f",
    "cmd": "lfortran --fixed-form --show-ast --no-color {infile} -o {outfile}",
    "infile": "tests/loop_test3.f",
    "infile_hash": "563e057e78a555b01f0ba784adcca08cd39669ac229fc2d62af5f13a",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "ast-loop_test3-f4e458f.stdout",
    "stdout_hash": "764d92558be0fd94ea39ed7aa4d15229b2b61a50973aa8a908ba76e3",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}