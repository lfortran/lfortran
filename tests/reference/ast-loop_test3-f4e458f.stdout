(TranslationUnit
    [(Program
        test
        ()
        []
        []
        []
        [(DoLoop
            0
            ()
            240
            k
            1
            mp
            ()
            [(=
                0
                (FuncCallOrArray
                    con
                    []
                    [(()
                    k
                    ())]
                    []
                    []
                )
                (u- (FuncCallOrArray
                    datmat
                    []
                    [(()
                    k
                    ())
                    (()
                    np
                    ())]
                    []
                    []
                ))
                ()
            )
            (DoLoop
                0
                ()
                220
                j
                1
                n
                ()
                [(=
                    220
                    (FuncCallOrArray
                        w
                        []
                        [(()
                        j
                        ())]
                        []
                        []
                    )
                    (+ (FuncCallOrArray
                        datmat
                        []
                        [(()
                        k
                        ())
                        (()
                        j
                        ())]
                        []
                        []
                    ) (FuncCallOrArray
                        con
                        []
                        [(()
                        k
                        ())]
                        []
                        []
                    ))
                    ()
                )]
                ()
                ()
            )
            (DoLoop
                0
                ()
                240
                i
                1
                n
                ()
                [(=
                    0
                    temp
                    (Real "0.0d0")
                    ()
                )
                (DoLoop
                    0
                    ()
                    230
                    j
                    1
                    n
                    ()
                    [(=
                        230
                        temp
                        (+ temp (* (FuncCallOrArray
                            w
                            []
                            [(()
                            j
                            ())]
                            []
                            []
                        ) (FuncCallOrArray
                            simi
                            []
                            [(()
                            j
                            ())
                            (()
                            i
                            ())]
                            []
                            []
                        )))
                        ()
                    )]
                    ()
                    ()
                )
                (If
                    0
                    ()
                    (== k mp)
                    [(=
                        0
                        temp
                        (u- temp)
                        ()
                    )]
                    []
                    ()
                    ()
                    ()
                )
                (=
                    240
                    (FuncCallOrArray
                        a
                        []
                        [(()
                        i
                        ())
                        (()
                        k
                        ())]
                        []
                        []
                    )
                    temp
                    ()
                )]
                ()
                ()
            )]
            ()
            ()
        )]
        []
    )]
)
